Distinct collaborators receiving the same task

**Describe the bug**
When two collaborators log into the same project they receive the same task
Backend indicates an error:
```
[2021-04-12 22:11:03,640] [tasks.models::set_lock::103] [ERROR] Current number of locks for task 2 is 1, but overlap=1: that's a bug because this task should not be taken in a label stream (task should be locked)
```
In the user screen that logged last, sometimes, the "Submit" button disappear.

**To Reproduce**
Steps to reproduce the behavior:
1. Create accounts "user1@localhost" and "user2@localhost"
2. Log user1 and create a sample project
3. Import file with multiple tasks
4. Click to label a task with user1 
5. Log user2 (using another browser or anonymous session)
6. Click to label a task with user2


**Expected behavior**
Users receiving different tasks each.

**Screenshots**
Click to label
![Captura de tela de 2021-04-13 14-16-39](https://user-images.githubusercontent.com/20566121/114730079-eb73ac80-9d16-11eb-9f1b-67ed2a05e4aa.png)

user2 disappearing top toolbar 
![Captura de tela de 2021-04-14 11-47-50](https://user-images.githubusercontent.com/20566121/114730466-473e3580-9d17-11eb-8f07-41da9e252e76.png)


**Environment (please complete the following information):**
 - OS: Ubuntu 20.04 focal
 - Label Studio Version 1.0.1

**EDIT**
Better description to steps to reproduce the issue and possibly related "Submit button disappear" issue reported above and by @bolotyuh

--- a/label_studio/projects/api.py
+++ b/label_studio/projects/api.py
@@ -280,6 +280,20 @@ class ProjectNextTaskAPI(generics.RetrieveAPIView):
             except Task.DoesNotExist:
                 logger.debug('Task with id {} locked'.format(task.id))
 
+    def _get_first_locked_by(self, user, tasks_query):
+        def match(task):
+            # Match task locked by user and discard expired tasks
+            return (
+                task.has_lock()
+                and task.locks.filter(user=user).count() > 0
+            )
+
+        lookup = (
+            task for task in tasks_query.all()
+            if match(task)
+        )
+        return next(lookup, None)
+
     def _try_ground_truth(self, tasks, project):
         """Returns task from ground truth set"""
         ground_truth = Annotation.objects.filter(task=OuterRef('pk'), ground_truth=True)
@@ -431,10 +445,14 @@ class ProjectNextTaskAPI(generics.RetrieveAPIView):
 
             # ordered by data manager
             if external_prepared_tasks_used:
-                next_task = not_solved_tasks.first()
+                use_task_lock = False
+                next_task = self._get_first_locked_by(user, not_solved_tasks)
+                if not next_task:
+                    use_task_lock = True
+                    next_task = self._get_first_unlocked(not_solved_tasks)
                 if not next_task:
                     raise NotFound('No more tasks found')
-                return self._make_response(next_task, request)
+                return self._make_response(next_task, request, use_task_lock=use_task_lock)
 
             # If current user has already lock one task - return it (without setting the lock again)
             next_task = Task.get_locked_by(user, project)

